configuration_version_integration_test.go

```diff
@@ -242,6 +242,73 @@ func TestConfigurationVersionsArchive(t *testing.T) {
 	client := testClient(t)
 	ctx := context.Background()
 
+	w, wCleanup := createWorkspace(t, client, nil)
+	defer wCleanup()
+
+	cv, cvCleanup := createConfigurationVersion(t, client, w)
+	defer cvCleanup()
+
+	t.Run("when the configuration version exists and has been uploaded", func(t *testing.T) {
+		err := client.ConfigurationVersions.Upload(
+			ctx,
+			cv.UploadURL,
+			"test-fixtures/config-version",
+		)
+		require.NoError(t, err)
+
+		// We do this is a small loop, because it can take a second
+		// before the upload is finished.
+		for i := 0; ; i++ {
+			refreshed, err := client.ConfigurationVersions.Read(ctx, cv.ID)
+			require.NoError(t, err)
+
+			if refreshed.Status == ConfigurationUploaded {
+				break
+			}
+
+			if i > 10 {
+				t.Fatal("Timeout waiting for the configuration version to be uploaded")
```

Comment:
I notice that there is a similar loop later in the code. Any chance of refactoring to avoid the duplication?

Also, I'm concerned about adding a test that might have flaky behavior. Specifically, I'm wondering where the 10 second wait comes from? What if we need 15 seconds for this to pass? Could other tests running in tflocal impact the performance of this test, for example?

Reply:
> I notice that there is a similar loop later in the code. Any chance of refactoring to avoid the duplication?

Fair enough! I'll deduplicate the code now.

> Also, I'm concerned about adding a test that might have flaky behavior. Specifically, I'm wondering where the 10 second wait comes from? What if we need 15 seconds for this to pass? Could other tests running in tflocal impact the performance of this test, for example?

Also fair! Unfortunately, I don't know anything we can do here besides increasing the timeout to something absurd like a full minute maybe? That feels super icky to me, but for this specific case it doesn't seem like we have any other alternatives, unless you're advocating for keeping the test deleted? I'm 50/50, heck I wasn't even the person who originally wrote this, so I imagine whoever that is they'd probably be the ones to ask.

Reply:
It'd be awesome if we could somehow materialize the promise to archive/upload the config versions so we could avoid all the polling, but since that archiving/uploading happens on a background job, we're kinda hosed there 😢

Reply:
You're right - a minute doesn't seem awesome. What do you think about adding a comment where the timeout is so future us knows that if we see some flakiness there it might be that we need a longer timeout?